### ISSUE TITLE
[Filter Panel]: QA: Screen reader is not identifying the checkboxes in the Filter panel

### ISSUE THREAD

**Issue by gopinathan-n** [carbon-design-system/carbon-for-ibm-dotcom#9296]
### Description

Page Element:
Check boxes present in the following filter options: 
Product types
Deployment types

Issue Description:
Following issue is found in JAWS with FF: 
1. in the product types option, Screen reader identifies only "API" check box in up/down arrow navigation. However in tab/shift tab navigation, all the check boxes are identified. Same issue is found in the check boxes present in the "Deployment types" filter option as well.

Expected:
"Screen reader is expected to identify all the checkboxes in arrow key navigation as well.
We have cross checked this issue in the following Screen reader / browser combinations and this issue is only reproduce able in jaws versions 2021,2020 and 2019 with FF
JAWS with Edge and chrome works fine
NVDA with FF/also works fine
VoiceOver iPhone with Safari works fine as well
Our FF version is 94"
Since jaws is IBM recommended Screen reader and FF is the IBM recommended browser for internal use, We may have to look for a possible fix for this issue.

User Impact:
When the screen reader doesn’t identify the page elements, it impacts the experience of users who are blind or have vision impairment. They can miss important information or fail to perform an action on the web page. 

### Component(s) impacted

Filter Panel

### Browser

Firefox

### Carbon for IBM.com version

v1.37.0-rc.0

### Severity

Severity 2 = Aspects of design is broken, and impedes users in a significant way, but there is a way to complete their tasks. Affects major functionality, has a workaround.

### Application/website

Carbon for IBM.com

### Package

@carbon/ibmdotcom-web-components

### CodeSandbox example

https://carbon-design-system.github.io/carbon-for-ibm-dotcom/canary/react-experimental/?path=/story/components-audio-player--default

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/.github/CODE_OF_CONDUCT.md)
- [ ] I checked the [current issues](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues) for duplicate issues

**Comment by andy-blum** [comment]
Closing in favor of #10589
